code-reviews/recipes/Terraform.md

```diff
@@ -53,10 +58,16 @@ echo -e "\n\n>>> Terraform validate"
 terraform validate
 ```
 
-> TODO: Add example of how to use this in an AzDO build validation pipeline
-
-## Code Review Checklist
-
+### Code Review Checklist
 In addition to the [Code Review Checklist](../readme.md) you should also look for these terraform specific code review items
 
-> TODO: Add checklist
+- Verify Terraform project is configured to use Azure Storage as remote state backend
+- Verify remote state backend storage account key is stored in the key vault and not hard coded
+- If Terraform code is mixed with application source code, verify that the Terraform code is isolated into a dedicated folder
+- If the infrastructure going to different depending on the environment (Example: Dev, UAT, Production), verify sub-directory per environment is created
+- Verify resource definition and data sources are handled correctly in the Terraform
+    - resource : Indicates to Terraform that the current configuration is in charge of managing the life cycle of the object
+    - data: Indicates to Terraform that you only want to get a reference of the existing object, but don’t want to manage it part of this configuration
+- Ensure code is split into to multiple reusable modules
+- Verify for unit test implementation for Terraform code. [Terratest](https://terratest.gruntwork.io/) is widely used. While running the unit tests from DevOps pipelines, check for managed identities or service principal is used with the Terratest
```

Comment:
reword suggestion: Verify that unit tests are used (E.g. Terratest). While running unit tests from DevOps pipelines, check that managed identities or service principals are used.

Reply:
Reworded based on the suggestion.